### PR TITLE
Fix 3.9 MacOS from-source build

### PR DESCRIPTION
**Pull request by jorjun**
sysconfig.get_config_var('MACOSX_DEPLOYMENT_TARGET')

returns <str> '11.1' in Python 2.7 and <int> 11 in Python 3.9 causing exception: builtins.TypeError: expected string or bytes-like object

@nicolasnoble

Fixes https://github.com/grpc/grpc/issues/25082

### CODE REVIEW

**Comment by ReSqAr** [comment]
My work around is to use:
```
export SYSTEM_VERSION_COMPAT=1
```
so the OS identifies as version 10.16, see the command `sw_vers`.

File: setup.py
Comment (ReSqAr) [inline setup.py:360]: I don't think this will work as intended - a `None` value for example will be converted to a string `'None'`. Maybe it's better to use:
```
pkg_resources.parse_version(str(mac_target))
```
which only gets executed after `if mac_target`.

Comment (gnossen) [reply setup.py:360]: Agreed. @jorjun I think this PR will be good to go if you make this change.

**Review by gnossen — APPROVED** [review]

**Comment by gnossen** [comment]
Adding @lidizheng for a second review.

**Review by lidizheng — APPROVED** [review]
This PR fixes https://github.com/grpc/grpc/issues/25082

**Comment by hateyou218** [comment]
When will this get released?

**Pull request merged**
